jobserver/models/outputs.py

```diff
@@ -158,6 +158,21 @@ def get_absolute_url(self):
 
     def get_api_url(self):
         """The API url that will serve up this file."""
+        if getattr(self, "is_published", False):
+            # This is a hack so the view for viewing a published ReleaseFile
+            # can pass down the knowledge that a ReleaseFile has been
+            # published, avoid O(N) queries on views which aren't expecting to
+            # have to add `.select_related("snapshot")` to a QuerySet.
+            return reverse(
+                "published-file",
+                kwargs={
+                    "org_slug": self.workspace.project.org.slug,
+                    "project_slug": self.workspace.project.slug,
+                    "workspace_slug": self.workspace.name,
+                    "file_id": self.id,
+                },
+            )
+
```

Comment:
Hmm, am not sure about this.

This means, if you view the *release* for this file, it could use the public url for some files (that are part of a published snapshot) and the private url for others.

That seems a bit odd, but I guess we don't really care, as it's all hidden away

Reply:
Yeah, I didn't really love how this panned out but because we just pass ReleaseFile instances down to `generate_index` there wasn't an obvious path.